base/inc/AdePT/BVHNavigator.h

```diff
@@ -44,13 +96,24 @@ class BVHNavigator {
 
     Vector3D currentpoint(point);
     Vector3D daughterlocalpoint;
+    long exclude_id = -1;
+    long vol_id     = -1;
 
     for (auto v = vol; v->GetDaughters().size() > 0;) {
       auto bvh = vecgeom::BVHManager::GetBVH(v->GetLogicalVolume()->id());
 
-      if (!bvh->LevelLocate(exclude, currentpoint, v, daughterlocalpoint)) break;
+      exclude_id = -1;
+      if (exclude != nullptr) {
+        exclude_id = exclude->id();
+      }
+      vol_id = -1;
+
+      if (!bvh->LevelLocate<CandidateContains, SkipItem, ItemId>(exclude_id, currentpoint, vol_id, daughterlocalpoint))
```

Comment:
See my comment about the template mechanism: https://gitlab.cern.ch/VecGeom/VecGeom/-/merge_requests/999#note_7095383